app/src/main/java/com/github/se/project/ui/overview/Overview.kt

```diff
@@ -0,0 +1,165 @@
+
+@Composable
+fun HomeScreen(
+    listProfilViewModele: ListProfilesViewModel,
+    lessonViewModel: LessonViewModel,
+    navigationActions: NavigationActions,
+) {
+
+  val context = LocalContext.current
+  val currentProfile = listProfilViewModele.currentProfile.collectAsState().value
+  val lessons = lessonViewModel.currentUserLessons.collectAsState().value
+
+  // lessonViewModel.getLessonsByUserId(currentProfile?.uid ?: "")
+  when (currentProfile?.role) {
+    Role.TUTOR -> lessonViewModel.getLessonsForTutor(currentProfile.uid)
+    Role.STUDENT -> lessonViewModel.getLessonsForStudent(currentProfile.uid)
+    Role.UNKNOWN -> Toast.makeText(context, "Unknown Profil", Toast.LENGTH_SHORT).show()
+    null -> Toast.makeText(context, "null Profil", Toast.LENGTH_SHORT).show()
+  }
+
+  // Determine which bottom navigation items to show based on the user's role
+  val LIST_TOP_LEVEL_DESTINATIONS =
+      when (currentProfile?.role) {
+        Role.TUTOR -> LIST_TOP_LEVEL_DESTINATIONS_TUTOR
+        Role.STUDENT -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+        Role.UNKNOWN -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+        null -> LIST_TOP_LEVEL_DESTINATIONS_STUDENT
+      }
+
+  Scaffold(
+      topBar = {
+        Row(
+            modifier = Modifier.fillMaxWidth().padding(16.dp),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically) {
+              IconButton(
+                  onClick = {
+                    Toast.makeText(context, "Profile icon clicked", Toast.LENGTH_SHORT).show()
+                    navigationActions.navigateTo(Screen.PROFILE)
+                  }) {
+                    Icon(imageVector = Icons.Filled.AccountBox, contentDescription = "Profile Icon")
+                  }
+            }
+      },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATIONS,
+            selectedItem = navigationActions.currentRoute())
+      },
+      content = { paddingValues ->
+        if (currentProfile == null) {
+          // Display a message when no profile is found
+          NoProfileFoundScreen(context, navigationActions)
+        } else if (lessons.isNotEmpty()) {
+          LazyColumn(modifier = Modifier.fillMaxWidth().padding(paddingValues)) {
+            items(lessons.size) { index ->
+              if (currentProfile != null) {
+                LessonItem(
+                    currentProfile,
+                    lesson = lessons[index],
+                    onClick = {
+                      // Logic for selecting a lesson and navigating to the edit screen
+                      lessonViewModel.selectLesson(lessons[index])
+                      // TODO: nagivate to edit lesson screen
+                      Toast.makeText(context, "Navigate to EDIT_LESSON screen", Toast.LENGTH_LONG)
+                          .show()
```

Comment:
same comment here,  we might want to remove the toast. as the edit lesson is note implemented yet, we can keep it for now!